src/Zynga/Framework/Type/V1/Base.hh

```diff
@@ -62,6 +65,27 @@ abstract class Base implements TypeInterface {
     }
   }
 
+  <<__Override>>
+  public function equals(mixed $value): bool {
+    if ($value === null) {
+      return false;
+    }
+
+    // This first line is because "$value instanceof self" will be true for
+    // everything inheriting from Base.hh
+    // What we REALLY want to do is check if $value is a subclass of
+    // $this's class, not that $value is a subclass of Base.hh
+    if (is_a($value, get_class($this)) &&
+    // This tells the stupid typechecker that the next line is kosher
+    // because it's too dumb to understand that the above line means that
```

Comment:
We should probably rephrase this and the line above so that we only specify the limitations of the type checker